libs/core/langchain_core/tools/base.py

```diff
@@ -479,10 +479,16 @@ def _parse_input(self, tool_input: Union[str, Dict]) -> Union[str, Dict[str, Any
             return tool_input
         else:
             if input_args is not None:
-                result = input_args.parse_obj(tool_input)
+                if issubclass(input_args, BaseModel):
+                    result = input_args.model_validate(tool_input)
+                    result_dict = result.model_dump()
+                else:
```

Comment:
we should probably do

```python
else issubclass(input_args, pydantic.v1.BaseModel): 
  ..
else:
  raise NotImplementedError() # or assertion error
```

to be very explicit about compatibility (so we can also grep for it later on in the code)